### PR TITLE
chore: Add warning when use `value` of Switch, Checkbox, Upload

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

ref #10945 #9611 #7481 #6261 #3142 #1550

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Add warning when use `value` of Switch, Checkbox, Upload      |
| 🇨🇳 Chinese |     Switch、Checkbox 和 Upload 使用 `value` 时会提示正确的属性名。       |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/checkbox/__tests__/checkbox.test.js [hunk 1]
```diff
@@ -2,8 +2,9 @@ import React from 'react';
 import { mount } from 'enzyme';
 import Checkbox from '..';
 import focusTest from '../../../tests/shared/focusTest';
+import { resetWarned } from '../../_util/warning';
 
-describe('Checkbox', () => {
+describe('Checkbox.2333', () => {
```
Comment (afc163) [inline components/checkbox/__tests__/checkbox.test.js]: 2333

File: components/switch/index.tsx [hunk 2]
```diff
@@ -43,6 +44,16 @@ export default class Switch extends React.Component<SwitchProps, {}> {
 
   private rcSwitch: typeof RcSwitch;
 
+  constructor(props: SwitchProps) {
+    super(props);
+
+    warning(
+      'checked' in props || !('value' in props),
```
Comment (afc163) [inline components/switch/index.tsx]: 只要是指定了 value 就 warning 吧，不管有没有 checked

Comment (zombieJ) [reply components/switch/index.tsx]: 为了保险一点，检查都是没填正确的 prop 且填了 value 才报错。免得误报，因为看我们自己测试用例也会用错。

File: components/checkbox/__tests__/checkbox.test.js (hunk 1 above)
Comment (zombieJ) [reply components/checkbox/__tests__/checkbox.test.js]: 😝

File: components/switch/index.tsx (hunk 2 above)
Comment (afc163) [reply components/switch/index.tsx]: 我是怕有人这么写

```jsx
getDecorator({ name: 'xxx', initialValue: true })(
  <Switch checked={this.state.checked} />
)
```

**Pull request merged**